### PR TITLE
feat: add --compat flag to dagger to specify api version

### PR DESCRIPTION
**Pull request by jedevc**
We removed this last minute in https://github.com/dagger/dagger/pull/7759/#discussion_r1668419158.

However, it does seem like we need *something* here - so proposing this back in. Let the bikeshedding of naming/etc begin, I don't have particularly strong opinions here :tada:

### CODE REVIEW

**Comment by helderco** [comment]
`--no-upgrade` could be an option too, to just not update that field. But good with `--compat <version>` as well since it's perhaps clearer what it's for.

Maybe allow `--compat [version]` (optional version) so that a standalone `--compat` means "don't update engineVersion", while specifying the version overrides the value for it.

I also suggest keeping it for `develop` only, instead of new modules.

**Comment by jedevc** [comment]
I'd kinda like to have it also for `init` since that can do codegen as well. I think it's definitely a *less* common usecase, but I definitely have it while playing around.

I like the sound of an optional version, to keep the current version.

I'll also add some tests.

**Comment by helderco** [comment]
> I'd kinda like to have it also for `init` since that can do codegen as well. I think it's definitely a _less_ common usecase, but I definitely have it while playing around.

How about this workaround then?
```
dagger init
dagger develop --compat --sdk=...
```

**Comment by jedevc** [comment]
@helderco done all the above.

Weird note - it needs to specified either as `--compat` or `--compat=<version>` - you can't use `--compat <version>`, see https://github.com/spf13/cobra/issues/866.

**Review by shykes — APPROVED** [review]

File: docs/static/api/reference/index.html
Comment (helderco) [inline docs/static/api/reference/index.html:1]: Is this now always producing a diff on generate due to random ids? Might make sense to discard changes if so.

**Review by helderco — APPROVED** [review]
lgtm 👍

File: docs/static/api/reference/index.html
Comment (jedevc) [reply docs/static/api/reference/index.html:1]: Hm, this was added in https://github.com/dagger/dagger/pull/7937. Yeah, these look random...

But what's super annoying is that even today these random examples will be re-updated on every single netlify deploy? We should definitely fix this.

cc @grouville (also, looks like we're missing this in docs linting, which would have caught this)

**Pull request merged**